### PR TITLE
Update Kafka broker version used in tests and add hostname to certificate

### LINKED ISSUE #19762: Update version of Kafka used for the Reactive Messaging tests
Our tests use the confluent platform docker images.

In some tests we're using 5.4.3 (kafka 2.4.3) and in other tests we're using 4.0.0 (kafka 1.0.0).

The correspondence between confluent plaform versions and kafka versions is listed on [this page][cp-kafka-versions].

When we update, we will probably have to re-do some of the work to enable authentication with the broker and deploy the necessary credentials and certificates as it's dependent on the structure of the older confluent platform docker images.

Kafka currently uses log4j v1 which is EOL since 2015, though they [don't list any known vulnerabilities][kafka-cves] unless the JMS appender is used.

[Work is currently underway][kafka-log4j2] for Kafka to move up to a supported version of log4j2, but this hasn't been completed yet.

[cp-kafka-versions]: https://docs.confluent.io/platform/current/installation/versions-interoperability.html#
[kafka-cves]: https://kafka.apache.org/cve-list
[kafka-log4j2]: https://issues.apache.org/jira/browse/KAFKA-9366

### PR DESCRIPTION
**Pull request by Azquelt**
## Update Kafka broker version

- Make KafkaTlsContainer and KafkaSaslPlainContainer common and use the
same container code across all tests.
- Refactor getting the connection properties, since this can no longer be
derived directly from the container.
- Update some older tests to use KafkaClientTestProvider
- Clean up some helper methods which are no longer used
- Fixes #19762

## Add hostname to test certificate

When given an IP address for the kafka server, try to do a reverse DNS
lookup to find the hostname and include the hostname in the test
certificate.

Somehow the kafka client sometimes seems to use the hostname when
checking the certificate, even when connecting using the IP address.

Fixes RTC 290493

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#libby

**Comment by Azquelt** [comment]
#build
#libby

**Comment by Azquelt** [comment]
Seeing slightly different behaviour with the new version of the broker. Sometimes we look for for the partitions for a topic while the topic is still being created and we get an empty list back. As this is in test framework code, I've just added a retry with a timeout.

#libby
#build

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/containers/ExtendedKafkaContainer.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
+package com.ibm.ws.microprofile.reactive.messaging.fat.kafka.containers;
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/containers/ExtendedKafkaContainer.java:24]: Missing copyright

Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/containers/ExtendedKafkaContainer.java:24]: Even though this was copied from `org.testcontainers.containers.KafkaContainer`, it should have something at the top that says what the original copyright & license is.

**Review by tevans78 — APPROVED** [review]

**Comment by Azquelt** [comment]
#build
#libby

**Pull request merged**